contracts/cosmwasm-vm/cw-ibc-core/src/ics04_channel/packet/send_packet.rs

```diff
@@ -82,17 +82,22 @@ impl<'a> CwIbcCoreContext<'a> {
             });
         }
 
-        cw_println!(deps, " check pass: packet exipred");
+        cw_println!(deps, " check pass: packet height expired");
 
-        let consensus_state_of_b_on_a =
-            self.consensus_state(deps.storage, client_id_on_a, &latest_height_on_a)?;
-        let latest_timestamp = consensus_state_of_b_on_a.timestamp();
         let packet_timestamp = to_ibc_timestamp(packet.timeout_timestamp)?;
-        if let Expiry::Expired = latest_timestamp.check_expiry(&packet_timestamp) {
-            return Err(PacketError::LowPacketTimestamp).map_err(Into::<ContractError>::into);
+        let client = self.get_client(deps.storage, client_id_on_a)?;
+        let timestamp_at_height = client.get_timestamp_by_height(
+            deps.as_ref(),
+            client_id_on_a,
+            latest_height_on_a.revision_height(),
+        )?;
+        if timestamp_at_height != 0 {
+            let chain_timestamp = to_ibc_timestamp(timestamp_at_height)?;
+            if let Expiry::Expired = chain_timestamp.check_expiry(&packet_timestamp) {
+                return Err(PacketError::LowPacketTimestamp).map_err(Into::<ContractError>::into);
+            }
+            cw_println!(deps, " timestamp check pass");
         }
```

Comment:
This should be fine but not that ibc spec does not care if timestamp is expired just that height is not expired or that it is 0. But both can't be zero

```
    abortTransactionUnless(timeoutHeight !== 0 || timeoutTimestamp !== 0)
    
    // check that the timeout height hasn't already passed in the local client tracking the receiving chain
    latestClientHeight = provableStore.get(clientPath(connection.clientIdentifier)).latestClientHeight()
    abortTransactionUnless(timeoutHeight === 0 || latestClientHeight < timeoutHeight)
```

Reply:
it does pass e2e because  we were only doing timeout check by height. But now if icon sends packet with timestamp >0 it should fail because icon lightclient will always return zero for timestamp.

Reply:
Okey good